field/src/field.rs

```diff
@@ -192,6 +192,8 @@ pub trait PrimeField64: PrimeField {
     const ORDER_U64: u64;
 
     // TODO: Move to Field itself? Limiting it to `PrimeField64` seems unusual.
+    #[must_use]
+    #[allow(clippy::cast_possible_truncation)]
```

Comment:
Hm I think I'd prefer if we stop using `allow()` for pedantic lints, just to avoid clutter. If it means we can't benefit as much from pedantic lints (because we no longer have an easy way to check for new ones), I think that may still be the preferred tradeoff, since the benefits from pedantic nits seem pretty minor.

Reply:
I agree in general, but I think it's justified in this case.  Clippy was actually complaining that on 32-bit machines the cast to `usize` might truncate.

The main problem here is that the `allow` is fairly generic, instead of the more specific thing we actually want to say "We know that the logarithm only gives small results that would fit in a `usize` even on an 8 bit machine, so please don't bother us."

However, I can remove this annotation from the PR, if you want to.

Reply:
Hm I agree it's a false positive so we're justified in leaving the code, just a question of when we should use `allow` to silence false positives vs just ignoring them. I think this should probably correspond to what's enforced in CI; if we don't think a lint is important enough to enforce then IMO it's probably not worth the clutter to suppress it.

Do you think `cast_possible_truncation` is one of the more important pedantic lints that we should consider enforcing? Seems reasonable especially if there's only one false positive so far.

Reply:
Yes, I was considering suggestion `cast_possible_truncation` as a lint to enforce, and this PR can be seen as a first step in that direction.

Or more nuanced: I'm not so much against casts that can possibly truncate, but those are dangerous enough that it's perhaps useful to force people to explicitly stick `#[allow(clippy::cast_possible_truncation)]` in the code, to make sure they are aware of what they are getting into.